modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/expression/FilterByCountIndividualSegmentsExpressionVisitorImpl.java

```diff
@@ -112,6 +114,61 @@ else if (Objects.equals("day", value)) {
 			"Unsupported operation with value " + value);
 	}
 
+	@Override
+	public FilterByCount.Day visitFunctionCallExpression(
+		@NotNull
+			IndividualSegmentsExpressionParser.FunctionCallExpressionContext
+				functionCallExpressionContext) {
+
+		ParseTree parseTree1 = functionCallExpressionContext.getChild(0);
+
+		Object functionName = parseTree1.accept(this);
+
+		if (Objects.equals("between", functionName)) {
+			ParseTree parseTree2 = functionCallExpressionContext.getChild(2);
+
+			List<String> parameters = (List<String>)parseTree2.accept(this);
+
+			if (parameters.size() != 3) {
+				throw new UnsupportedOperationException(
+					StringBundler.concat(
+						"Unsupported function BETWEEN ",
+						String.valueOf(parameters.size()), " params"));
+			}
+
+			if (!Objects.equals("day", parameters.get(0))) {
+				throw new UnsupportedOperationException(
+					"Unsupported function BETWEEN with first param " +
+						String.valueOf(parameters.get(0)));
+			}
+
+			return new FilterByCount.Day(
+				StringUtil.lowerCase(Function.BETWEEN.toString()),
+				parameters.get(1), parameters.get(2));
+		}
+
+		throw new UnsupportedOperationException(
+			"Unsupported function " + functionName);
+	}
+
+	@Override
+	public List<String> visitFunctionParameters(
+		@NotNull IndividualSegmentsExpressionParser.FunctionParametersContext
+			functionParametersContext) {
+
+		List<String> parameters = new ArrayList<>();
```

Comment:
if we return an array of parameters we can use them directly in the  FilterByCount.Day constructor, right?

Reply:
No, because in the first parameter we have the day variable.

Reply:
No, using the same for loop, but creating moving parameters from List<String> parameters to String[] parameters

Reply:
ok, no problem for that. I change the List to String[].

Reply:
Forget my last comment, I was thinking in other things. I am going to change from list to String[]